src/main/java/reposense/parser/PathLocationParser.java

```diff
@@ -0,0 +1,71 @@
+package reposense.parser;
+
+import static reposense.util.FileUtil.fileExists;
+
+import java.nio.file.Paths;
+import java.util.Arrays;
+
+/**
+ * Contains functionality to parse repo location details from a path to a repo.
+ */
+public class PathLocationParser {
+
+    private static final String GIT_LINK_SUFFIX = ".git";
+    private static final String BRANCH_DELIMITER = "#";
+    /**
+     * Parses a given path to a repo and returns an array containing the following info:
+     * { path to repo, repository name, organisation name, branch name (if any) }
+     *
+     * @param pathAndBranch A path to a repository, with a branch name optionally appended at the end
+     *         e.g. /home/users/tom/Desktop/reposense#release
+     * @param isPathValidationNeeded If this flag is set to true, then the method will check
+     *         verify that {@code path} is a valid path and refers to an actual directory on the
+     *         filesystem.
+     *
+     * @return null if the given String is an invalid path, or no directory exists at the path,
+     *         and {@code isPathValidationNeeded} was set to true.
+     */
+    public static String[] tryParsingAsPath(String pathAndBranch, boolean isPathValidationNeeded)  {
+        String pathToRepo;
+        String branch;
+
+        if (fileExists(pathAndBranch)) {
+            // If the given path represents an existing file, then any # characters that occur in the path
+            // are part of the filenames in the path, rather than an indication of a branch name.
+            pathToRepo = pathAndBranch;
+            branch = null;
+        } else {
+            String[] split = splitPathAndBranch(pathAndBranch);
+            pathToRepo = split[0];
+            branch = split[1];
+            if (isPathValidationNeeded && !fileExists(pathToRepo)) {
+                return null;
+            }
+        }
+
+        String repoName = Paths.get(pathToRepo).getFileName().toString().replace(GIT_LINK_SUFFIX, "");
+        return new String[] { pathToRepo, repoName, null, branch };
+    }
+
+    /**
+     * Given a string of the form stringA#stringB, breaks the string at the # character
```

Comment:
Would this Javadoc be providing too much info? Usually we only document what and why aspects of the code, rather than how.

Perhaps `Returns a string array containing the {@code pathAndBranch} that is split based on the last '#' character`? or something like this would suffice?

Reply:
Thanks for the feedback. I actually think the javadoc is appropriate, and I have not tried to document *how* the method works. The description of *what* the method does could be summarised - I will do this.